### ISSUE TITLE
feat: semi-fungible-token to implement `transfer` without optional memo and `transfer-memo` with memo

### ISSUE THREAD

**Issue by fiftyeightandeight** [alexgo-io/alex-v1#352]
`trait-semi-fungible-token` to be refactored to implement the following:
https://github.com/MarvinJanssen/stx-semi-fungible-token/blob/a2504bb0eb0b4b340e30029e6823bb9e98ac97dc/contracts/sipxxx-semi-fungible-token-trait.clar#L21-L31

**Comment by SaadTahirTintash** [comment]
`sipxxx-semi-fungible-token-trait` This trait is different from what you've implemented. I think you've added more functions in it. I am just using the trait's new functions and keeping yours as well.

**Comment by SaadTahirTintash** [comment]
I've seen your usage of `sft-trait` and you are using `transfer-fixed` everywhere `(transfer-fixed (uint uint principal principal (optional (buff 34))) (response bool uint))` instead of `(transfer (uint uint principal principal) (response bool uint))` from the trait itself. Moreover, there is no place where you are sending any memo, every memo is set to `none`

**Comment by fiftyeightandeight** [comment]
Yes, the existing "transfer" and "transfer-fixed" need to be updated to reflect the trait (ie no memo). New "transfer-memo" and "transfer-memo-fixed" need to be created to allow transfer with memo.

**Comment by SaadTahirTintash** [comment]
The `get-decimals` in the `sipxxx-semi-fungible-token-trait` has signature `(get-decimals (uint) (response uint uint))` which is different from your implementation of `(get-decimals () (response uint uint))`.

**Comment by fiftyeightandeight** [comment]
@MarvinJanssen I was looking at get-decimals. The way it works currently is sum of supply of tokens = supply of semi-fungible token.
If each token has different decimals, it's not obvious to me how we can always ensure the equality.
I also see no obvious reasons why one may want different decimals for different tokens within a single semi-fungible token.
Anything I missed?

**Comment by SaadTahirTintash** [comment]
I don't see any implementation related to `transfer-many` and `transfer-many-memo` so I'm removing them from the trait for now.

**Comment by MarvinJanssen** [comment]
@fiftyeightandeight Not all SFTs have to be equal internally. Imagine a game like Moonray, they might represent their in-game currency as ID = 1, decimals = 6, but also have a different commodity as ID = 2, decimals = 0. It depends on the developer, in this case you could always return the same value and ignore the `uint` input parameter.

**Comment by fiftyeightandeight** [comment]
Ok, let's keep the flexibility. @SaadTahirTintash would you please update "get-decimals" to accept uint (but returns u8 always).
